.github/workflows/build.yml

```diff
@@ -131,28 +138,106 @@ jobs:
     - name: Publish NuGet packages to GitHub Packages
       run: dotnet nuget push "*.nupkg" --api-key ${{ secrets.GITHUB_TOKEN }} --skip-duplicate --no-symbols --source https://nuget.pkg.github.com/${{ github.repository_owner }}/index.json
 
-  # TODO Add Authenticode signing step before publish-nuget.
-  # See https://github.com/dotnet/sign/blob/main/docs/gh-build-and-sign.yml.
+  sign:
+    needs: publish-github
+    permissions:
+      id-token: write
+    runs-on: windows-latest
+    # TODO Commented-out for testing signing in a branch without actually tagging
+    #if: |
+    #  github.event.repository.fork == false &&
+    #  startsWith(github.ref, 'refs/tags/')
+
+    steps:
+
+    - name: Download unsigned packages
+      uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+      with:
+        name: packages-windows
+        path: packages
+
+    - name: Download signing configuration
+      uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+      with:
+        name: signing-config
+        path: signing-config
+
+    - name: Setup .NET SDK
+      uses: actions/setup-dotnet@3447fd6a9f9e57506b15f895c5b76d3b197dc7c2 # v3.2.0
+
+    - name: Install Sign CLI tool
+      run: dotnet tool install --tool-path . sign --version 0.9.0-beta.23127.3
+
+    - name: Azure CLI login
+      uses: azure/login@92a5484dfaf04ca78a94597f4f19fea633851fa2 # v1.4.7
+      with:
+        allow-no-subscriptions: true
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
```

Comment:
These secret names are just copy-pasted from the sample - happy to change them to something else that's more obvious what they're for.